dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -1,820 +1,397 @@
 
-When requests aren’t transactional, we’d need to build the POST & GET workflow into the client, adding complexity to what could have been simple.
-
-Make sure this is the case by, for example,  [setting \`refresh: wait\_for\`](https://github.com/elastic/kibana/blob/188669773e9cc1981b5e89f24afcfdc379ad3058/x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.ts#L644-L657) on the relevant indices in POST requests.
-
-```ts
-private async persistAlertsHelper() {
-  ...
-  await resolveAlertConflicts({
-    logger: this.options.logger,
-    esClient,
-    bulkRequest: {
-      refresh: 'wait_for', // Ensure the index is refreshed after the operation
-      index: this.indexTemplateAndPattern.alias,
-      require_alias: !this.isUsingDataStreams(),
-      operations: bulkBody,
-    },
-    bulkResponse: response,
-    ruleId: this.options.rule.id,
-    ruleName: this.options.rule.name,
-    ruleType: this.ruleType.id,
-  });
-  ...
-}
+**Cursor based response**
 
+```json
+{
+  "items": [...],
+  "total": 100,
+  "pit": "abc",
+  "search_after": "valueA1.valueB1",
+  "size": 10
+}
 ```
 
-### Offer compare-and-swap
+**Pagination requires sorting**
 
-Between refreshing state and applying changes, there’s a gap where someone could modify your API. Generally it's ok to take the approach of last-write-wins.
+Carefully decide a default for sorting resources. It is possible to accept a custom
+set of additional values for sorting, but this may not be needed. Adding this later is a non breaking change!
 
-If you have to consider concurrency control, support mechanisms like ETags and checksums, and the \`version\` property on saved objects.
+Specify sorting in the query parameters like:
 
-### Ignore the robustness principle: don’t normalize output
+```
+GET /api/my_resources?sort=field_name,-other_name
+```
 
-The robustness principle says to accept input in various formats but return output in a consistent format, for example converting strings to lowercase, ordering elements in a list or changing whitespace in \`json\` strings.
+Where `field_name` is the name of the field to sort by in ascending order and `-other_name` is the name of the field to sort by in descending order.
 
-For Terraform, don’t do this.
+### Filtering
 
-Terraform does byte-for-byte comparisons, so normalized output forces provider developers to implement logic to handle different data formats, ordering, capitalization variations and other unnecessary complexity.
+`GET /api/my_resources?filter=field_name:value`
 
-Fortunately, the kibana client hasn’t needed to handle complex normalization yet. Let’s keep it that way, return data exactly as you received it.
-Follow these tips, and you'll make your API a dream to work with for Terraform provider developers.
+Use [simple KQL](https://www.elastic.co/docs/explore-analyze/query-filter/languages/kql) to cover most of your filtering needs.
```

Comment:
What do folks think? Is this a good general recommendation? Should we encourage a different way of specifying filters?